### PR TITLE
[Merged by Bors] - fix: bugs in linarith preprocessing

### PR DESCRIPTION
**Pull request by digama0**
As [reported on Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/linarith.20failure/near/319509822).

### CODE REVIEW

File: test/linarith.lean [hunk 1]
```diff
@@ -40,6 +40,14 @@ example (A B : Rat) (h : 0 < A * B) : 0 < 8*A*B := by
 example (A B : Rat) (h : 0 < A * B) : 0 < A*8*B := by
   linarith
 
+example [LinearOrderedCommRing α] (x : α) : 0 ≤ x := by
+  have h : 0 ≤ x := sorry
```
Comment (gebner) [inline test/linarith.lean:44]:
```suggestion
example [LinearOrderedCommRing α] (x : α) (h : 0 ≤ x) : 0 ≤ x := by
```

File: test/linarith.lean [hunk 2]
```diff
@@ -40,6 +40,14 @@ example (A B : Rat) (h : 0 < A * B) : 0 < 8*A*B := by
...
+  linarith
+
+example [LinearOrderedCommRing α] (x : α) : 0 ≤ x := by
+  have h : 0 ≤ x := sorry
```
Comment (gebner) [inline test/linarith.lean:48]:
```suggestion
example [LinearOrderedCommRing α] (x : α) (h : 0 ≤ x) : 0 ≤ x := by
```
Let's get rid of the warnings!

**Review by gebner — COMMENTED** [review]
bors d+

File: test/linarith.lean (hunk 1 above)
Comment (hrmacbeth) [reply test/linarith.lean:44]: @gebner The bugfix tests the difference between these two things -- your version already passed.

Comment (gebner) [reply test/linarith.lean:44]: :bulb: Oh, they're testing the `withMainContext`!  That's cryptic..

**Pull request closed without merging**